### DISCUSSION TITLE
Blazer WASM / ReactiveUI

### DISCUSSION THREAD

**Discussion by mpinkus** [Marfusios/websocket-client#78] [General]
I'm working fine in a reactive viewmodel test, but in wasm I dont get the messages. websocket starts, no messages. Seems like a threading issue... any example would be appreciated for the wasm threading setup:

WebsocketClient = new WebsocketClient(this.AccessUriApp.AccessUri);
                    
                    WebsocketClient.ReconnectionHappened.Subscribe(info =>
                        this.Log().Debug($"PubSub Reconnection: {info.Type}"
                        )
                    );

                    WebsocketClient.MessageReceived
                        .Select(z => Observable.FromAsync(
                            async () => {
                            switch (z.MessageType)
                            {
                                case System.Net.WebSockets.WebSocketMessageType.Binary:
                                    this.Log().Debug($"%%%%%%%%%%%%%%% PubSub [Binary] Message Received %%%%%%%%%%%%%%%%%");

                                    break;
                                case System.Net.WebSockets.WebSocketMessageType.Close:
                                    this.Log().Debug($"%%%%%%%%%%%%%%% PubSub [Close] Message Received %%%%%%%%%%%%%%%%%");
                                    break;
                                case System.Net.WebSockets.WebSocketMessageType.Text:
                                    this.Log().Debug($"%%%%%%%%%%%%%%% PubSub [Text] Message Received %%%%%%%%%%%%%%%%%");
                                            break;

                                default:
                                    break;
                            }
                        }))
                        .Concat() // executes sequentially
                        .Subscribe();

                    _ = WebsocketClient.Start();

**Comment by mpinkus** [comment]
ok the issue is this: Disconnection: Error Operation is not supported on this platform.    at System.Net.WebSockets.ClientWebSocketOptions.set_KeepAliveInterval(TimeSpan value) ... any thoughts on use in wasm?

**Reply by Marfusios** [reply]
Hello @mpinkus, 

only `KeepAlive` configuration is not supported in Blazor? 
According to [this](https://gist.github.com/SteveSandersonMS/5aaff6b010b0785075b0a08cc1e40e01), `ClientWebsocketClient` should work in wasm.

**Reply by Marfusios** [reply]
Btw here is a working example: [crypto-watcher-blazor](https://github.com/Marfusios/crypto-watcher-blazor)
